packages/gen-ai-hub/src/utils/deployment-cache.test.ts

```diff
@@ -0,0 +1,127 @@
+import { type AiDeployment } from '@sap-ai-sdk/ai-core';
+import { deploymentCache } from './deployment-cache.js';
+import { FoundationModel } from './model.js';
+
+describe('deployment cache', () => {
+  afterEach(() => {
+    deploymentCache.clear();
+  });
+
+  it('should cache the deployment', () => {
+    const opts = {
+      scenarioId: 'foundation-models',
+      executableId: 'execution-id',
+      model: { name: 'gpt-4o', version: 'latest' }
+    };
+    deploymentCache.set(opts, {
+      id: 'deployment-id',
+      details: {
+        resources: {
+          backendDetails: { model: { name: 'gpt-4o', version: 'latest' } }
+        }
+      }
+    } as unknown as AiDeployment);
+
+    expect(deploymentCache.get(opts)).toEqual({
+      id: 'deployment-id',
+      model: { name: 'gpt-4o', version: 'latest' }
+    });
+  });
+
+  it('should cache all deployments independent of potentially given models', () => {
+    const opts = {
+      scenarioId: 'foundation-models',
+      model: { name: 'gpt-4o', version: 'latest' }
+    };
+
+    deploymentCache.setAll(opts, [
+      mockAiDeployment('deployment-id1', {
+        name: 'gpt-35-turbo',
+        version: 'latest'
+      }),
+      mockAiDeployment('deployment-id2', {
+        name: 'gpt-35-turbo',
+        version: '123'
+      })
+    ]);
+
+    expect(deploymentCache.get(opts)).toBeUndefined();
+    expect(
+      deploymentCache.get({
+        ...opts,
+        model: {
+          name: 'gpt-35-turbo',
+          version: 'latest'
```

Comment:
I would expect the test to pass with the following change, but it fails:

```suggestion
          name: 'gpt-35-turbo'
```